### PR TITLE
Fix Profiler tests and validation errors

### LINKED ISSUE #6822: Update profiler workflow based on #6782
## Context
We are currently passing the profiler metrics into a `TableProfile` in `core.py`. This include both column and table profile. 
https://github.com/open-metadata/OpenMetadata/blob/78b5f8c8e28ee78807e16dc0a5a2f3235e788c89/ingestion/src/metadata/orm_profiler/profiler/core.py#L423-L430
which are then passed into a `ProfilerResponse` object in the`profile` attribute.
https://github.com/open-metadata/OpenMetadata/blob/78b5f8c8e28ee78807e16dc0a5a2f3235e788c89/ingestion/src/metadata/orm_profiler/profiler/core.py#L381-L385
This `ProfilerResponse` object is then used in the sink to write data to the `tableProfile` attribute of the `Table` object.
https://github.com/open-metadata/OpenMetadata/blob/78b5f8c8e28ee78807e16dc0a5a2f3235e788c89/ingestion/src/metadata/orm_profiler/sink/metadata_rest.py#L70-L72
The sink uses `ingest_table_profile_data` method to add the data to the `Table` object
https://github.com/open-metadata/OpenMetadata/blob/78b5f8c8e28ee78807e16dc0a5a2f3235e788c89/ingestion/src/metadata/ingestion/ometa/mixins/table_mixin.py#L99-L113

With #6782 column profile and table are now split. table profile will be at the root of a table entity while column profile will be linked to table entities via the column entity (now accessible via `Table -> columns[Column -> profile]`.

## Solution
We need to integrate this new logic to the new logic of the profiler workflow.

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

Worked on fixing profiler model and separation of table and column api changes 
fixes #6822 

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/ingestion/sink/metadata_rest.py [hunk 1]
```diff
@@ -302,10 +305,12 @@ def write_tables(self, db_schema_and_table: OMetaDatabaseAndTable):
...
-            if db_schema_and_table.table.tableProfile is not None:
-                self.metadata.ingest_table_profile_data(
-                    table=created_table,
-                    table_profile=db_schema_and_table.table.tableProfile,
+            if db_schema_and_table.table.profile is not None:
+                self.metadata.ingest_profile_data(
+                    table=db_schema_and_table.table,
+                    profile_request=CreateTableProfileRequest(
+                        tableProfile=db_schema_and_table.table.profile
+                    ),
```
[the remark this answers is not in the document]
Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/sink/metadata_rest.py:313]: 🤔 if this sink is used for metadata migration, we might need it then..

[the remark this answers is not in the document]
Comment (TeddyCr) [reply ingestion/src/metadata/ingestion/sink/metadata_rest.py:313]: Oh I see, for users migrating from Amundsen to OM for example?

**Review by TeddyCr — APPROVED** [review]
@ayush-shah thanks for the PR, once the table test for the profile is fixed we can merge it. 🙏

File: ingestion/src/metadata/ingestion/sink/metadata_rest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmbrull) [reply ingestion/src/metadata/ingestion/sink/metadata_rest.py:313]: actually I believe this is used for the sample data. Amundsen/Atlas do not ship profiling

**Pull request merged**